cloudify_agent/api/pm/initd.py

```diff
@@ -280,6 +286,27 @@ def _create_config(self):
         self.runner.sudo('cp {0} {1}'.format(rendered, self.config_path))
         self.runner.sudo('rm {0}'.format(rendered))
 
+    def _create_start_on_boot_entry(self):
+
+        response = self.runner.run('which dpkg', exit_on_failure=False)
+        if response.code == 0:
+            # debian based distribution, run update-rc.d
+            self.runner.sudo('update-rc.d {0} defaults'.format(self.name))
+        else:
```

Comment:
This is kind of not scalable if we decide to add more distros:
Should it be more like:

```python
if self.runner.run('which dpkg', exit_on_failure=False) == 0:
    self.runner.sudo(...)
    return
elif self.runner.run('which rpm', exit_on_failure=False) == 0:
   self.runner.sudo(...)
   return
else:
   raise ...
```

?